### ISSUE TITLE
Is it GUI-d or bad to freeze sometimes?

### ISSUE THREAD

**Issue by KriSun95** [pet00184/flarepred#25]
Item 2 from #23: There is an issue where the GUI will freeze for unknown reasons and the bug seems intermittent. Is it due to wifi dropping? The user's computer going to sleep? Only time will tell!

This is a difficult on to describe since the bug is intermittent but is partly logged here in the accepted PR #22. The bugs are also recorded via the following two text files (not great, but all I have),
1. [flarePred_error2023July14.txt](https://github.com/pet00184/flarepred/files/12468604/flarePred_error2023July14.txt)
2. [flarePred_error2023July18_anotherFreeze.txt](https://github.com/pet00184/flarepred/files/12468605/flarePred_error2023July18_anotherFreeze.txt)

Both of these might be caused by the same issue or might be completely separate. They both share a common issue with the data handling, it seems, but that might be where the commonalities end. Both issues can be resolved in real time by hitting `Ctrl+C` in the terminal running the GUI which seems to restart things but this isn't ideal.

**Comment by KriSun95** [comment]
Just coming off the call, I ran the GUI and turned off my WIFI. The GUI became completely unresponsive and still wouldn't work after turning WIFI back on, it only worked once I `Ctrl-C`'d in the terminal.

This might not be the only reason this freezing happens but seems to be able to cause it. Might be worth checking on other machines and at other times. I hope there is a nice *Pythonic* way to fix it and, if not, we always have `try` and `except`...

**Comment by KriSun95** [comment]
Looking into this more, the common line in the package that is tripping up is 
```
File "/Users/kris/Documents/umnPostdoc/projects/both/foxsi4/flarePrediction/flarepred/RealTime/GOES_data_upload.py", line 50, in load_realtime_XRS
    wget.download(json_url, bar=None)
```

This seems to freak out anytime the internet connection is interrupted causing the GUI to hang. 

The example given [here](https://unogeeks.com/python-wget/) and some examples [here](https://www.programcreek.com/python/example/83386/wget.download) suggests using `try` and `except` statements with `wget.download` but none of them say why exactly.

I was hoping the `wget.download()` function would have a timeout option which might have helped out a bit or have been a valid fix; however, looking at the `wget.py` code the only options are to pass the URL, an output, and a downloading indicator
![Screenshot 2023-09-15 at 2 19 12 PM](https://github.com/pet00184/flarepred/assets/43237137/ad5fe621-d584-4fdb-acab-b827e01796e7)
but it does look like handling broken connections is on the list of things `wget.py` will be able to do...
![Screenshot 2023-09-15 at 2 19 29 PM](https://github.com/pet00184/flarepred/assets/43237137/6b0a6fdc-60f8-48ae-82dc-fa4eae9bf184)

The fix seems to be my initial guess with the `try` and `except` blocks with an extra step, the `except` block will call the function it's in (`GOES_data_upload.load_realtime_XRS()`) if the `wget` fails or is interrupted by, for example, the internet dropping. 

This fix means the GUI and any code relying on the download will still freeze up but will come back alive when everything is restored. The other option is to then write a good deal more code to store previous downloaded data and then use that until the, say, internet comes back on but this seems like a good bit of extra work and slightly deceptive since we might want to know when we're not getting any new data.

The internet dropping related freeze is addressed in #31; however, there may be other freeze conditions lurking about...

**Comment by KriSun95** [comment]
Question to ask, does the post analysis still work after the freeze?
